pypeit/core/wavecal/autoid.py

```diff
@@ -413,32 +430,56 @@ def load_linelist(self):
         self._wvdata.sort()
         return
 
-    def set_grids(self):
+    def set_grids(self, ngridw=200, ngridd=2000):
         # Set the wavelength grid
         if self._binw is None:
             # Ideally, you want binw to roughly sample the A/pix of the spectrograph
-            self._ngridw = 200
+            self._ngridw = ngridw
             self._binw = np.linspace(np.min(self._wvdata), np.max(self._wvdata), self._ngridw)
         else:
             self._ngridw = self._binw.size
         # Set the dispersion grid
         if self._bind is None:
-            self._ngridd = 2000
+            self._ngridd = ngridd
             self._bind = np.linspace(-3.0, 1.0, self._ngridd)
         else:
             self._ngridd = self._bind.size
         return
 
-    def run(self):
-        """Run through the parameter space and determine the best solution
-        """
-
+    def run_brute_loop(self, slit, arrerr=None, wavedata=None):
         # Set the parameter space that gets searched
         rng_poly = [3, 4]            # Range of algorithms to check (only trigons+tetragons are supported)
         rng_list = range(3, 10)      # Number of lines to search over for the linelist
         rng_detn = range(3, 10)      # Number of lines to search over for the detected lines
         rng_pixt = [1.0]             # Pixel tolerance
 
+        best_patt_dict, best_final_fit = None, None
+        # Loop through parameter space
+        for poly in rng_poly:
+            for lstsrch in rng_list:
+                for detsrch in rng_detn:
+                    for pix_tol in rng_pixt:
+                        psols, msols = self.results_brute(poly=poly, pix_tol=pix_tol, detsrch=detsrch, lstsrch=lstsrch,
+                                                          wavedata=wavedata, arrerr=arrerr)
+                        patt_dict, final_fit = self.solve_slit(slit, psols, msols)
+                        if final_fit is None:
+                            # This is not a good solution
```

Comment:
Should we add some `msgs` in here?

Reply:
Probably not - this is just looping through a bunch of parameter space and will only confuse the user, I think.